client/src/context/useNotificationContext.tsx

```diff
@@ -0,0 +1,41 @@
+import { Notification } from '../interface/Notification';
+import { fetchNotifications } from '../helpers/APICalls/getNotifications';
+import { updateNotifications } from '../helpers/APICalls/updateNotifications';
+import { createContext, FunctionComponent, useState, useContext, useEffect } from 'react';
+
+interface INotificationContext {
```

Comment:
```suggestion
interface NotificationContext {
```
'I' prefix is not necessary. It was a controversial topic, and you can read a lengthy discussion [here](https://github.com/microsoft/TypeScript-Handbook/issues/121).